services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/user_services_preferences/_packaging.py

```diff
@@ -0,0 +1,50 @@
+from collections.abc import AsyncIterator
+from contextlib import asynccontextmanager
+from pathlib import Path
+from typing import Final
+
+from pydantic import ByteSize, parse_obj_as
+from servicelib.archiving_utils import archive_dir, unarchive_dir
+from servicelib.file_utils import get_temporary_path_name, remove_directory
+
+from ._errors import DestinationIsNotADirectoryError, PreferencesAreTooBigError
+
+_MAX_PREFERENCES_TOTAL_SIZE: Final[ByteSize] = parse_obj_as(ByteSize, "128kib")
+
+
+@asynccontextmanager
+async def _temporary_path_name() -> AsyncIterator[Path]:
```

Comment:
What about simply doing this? Please read carefully https://docs.python.org/3/library/tempfile.html#tempfile.mkdtemp

```python

@asynccontextmanager
async def _temporary_path_name(suffix=None, prefix=None, dir=None) -> AsyncIterator[Path]:
     try:
        # NOTE: that the name of the directory is already unique
        temporary_dir = Path( tempfile.mkdtemp(suffix, prefix, dir) )
        yield temporary_dir / "archive_path"
    finally:
       await remove_directory(temporary_dir)
```

Reply:
in any case, I would avoid this helper and directly do something like
https://github.com/Tinche/aiofiles#aiofiles-file-support-for-asyncio

```python
async with aiofiles.tempfile.TemporaryDirectory() as tmp_dir:
     archive_path = Path(tmp_dir) / "archive"
     await archive_dir(source, archive_path, compress=True, store_relative_path=True)
     # ...
     return archive_path.read_bytes()
```

Reply:
and why did I not do this obvious last one I wonder.